rfc#00003.md

```diff
@@ -0,0 +1,221 @@
+
+### Applying trust domains to Enarx - default model
+There are various possible models for trust domains in an Enarx
+deployment - this section describes the simplest, and default model.
+In the default Enarx model, we start off by defining the following parties
+- tenant
+- host owner (the host operator may be separate, but is considered to
+be the same for this description)
+- entities running processes on the host (includes other workloads
+and compromised processes)
+- man-in-the-middle entities
+Clearly, the first two are the key actors - the others are included for
+reference (no threat model is defined here).
+
+The following components are defined:
+- host - the machine running workloads, including all hardware, firmware
+and software which is not specifically a workload
+- host CPU + CPU firmware (a subset of the host)
+- Orchestrator (e.g. Kubernetes, Openshift, Openstack)
+- Tenant workload image
+- Tenant workload image repository
+- Enarx client agent (ECA)
+- Enarx host agent (EHA)
+- Enarx runtime image
+- TEE instance
+- Empty Keep
+- Provisioned Keep
+- Tenant workload
+- Other workloads
+Not all of these exist at all points in the process.
+
+
+### Non-Enarx-enabled use case states
+In the non-TEE-assisted case, all processes running on the host are in
+the host's trust domain (this assumes no compromised processes - if
+there were any compromised processes, these would be in another, also
+untrusted domain).  We ignore man-in-the-middle entities.  For the
+default trust domain model, we assume that the Orchestrator is
+operated by the tenant and that the Tenant workload image repository
+is also operated by the tenant.
+
+#### State 0 non-Enarx-enabled use case (initial)
+No Enarx-related components are relevant to this use case.
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Tenant workload image
+  - Tenant workload image repository
+
+Note - there are models where a trust relationship is maintained by
+the tenant to the Tenant workload image repository or to the tenant
+workload, but it is not directly within the trust domain.  These
+models are not examined in this document.
+
+#### State 1 non-Enarx-enabled use case (provisioned)
+- host owner trust domain
+  - host
+  - host CPU + CPU firmware
+  - Tenant workload
+  - Other workloads
+- tenant trust domain
+  - Orchestrator
+  - Tenant workload image
+  - Tenant workload image repository
+  
+The change from state `0` to `1` is the provisioning of the Tenant workload
+onto the host from the Tenant workload image.
+
+The key point of this use case from a trust domain point of view it that
+since the host has access to all processes and associated memory that are
+running on it, all workloads that run on the host can only exist within
+the host trust domain.
+
+### Enarx use case states
+Note: that exact state details may vary by implementation.
```

Comment:
Instead of describing the state machine with text, I think the states that follow this heading should be represented in a diagram that accompanies this RFC.

Reply:
The `name` test now allows for Graphviz source to be included as well. FYI, @MikeCamel